File: py/solns/binaryTreeInorderTraversal/binaryTreeInorderTraversal.py
from solns.treeNode.treeNode import *
class Solution:
    @staticmethod
    def naive(root):
        res = []
        def inOrder(tree):
            if tree==None: return
            inOrder(tree.left)
            res.append(tree.val)
            inOrder(tree.right)
        inOrder(root)
        return res

File: cpp/example/src/BinaryTreeInorderTraversal/binaryTreeInorderTraversal.cpp
#include "binaryTreeInorderTraversal.h"

void binaryTreeInorderTraversal::recursive(TreeNode* tree){
    if (tree==nullptr) return;
    recursive(tree->left);
    this->res.push_back(tree->val);
    recursive(tree->right);
}

vector<int> binaryTreeInorderTraversal::naive(TreeNode* root){
    recursive(root);
    return this->res;
}
